java_src/foam/dao/MapDAO.java

```diff
@@ -50,7 +50,7 @@ public FObject remove(FObject obj) {
   }
 
   public FObject find(Object o) {
-    Object id = getOf().getClass().isInstance(o) || false ? getPrimaryKey().get(o) : o;
+    Object id = getPrimaryKey().get(o);
```

Comment:
This change is breaking other people. find() can take either a primarykey or an object.